src/persist/src/indexed/encoding.rs

```diff
@@ -205,7 +206,7 @@ pub struct BlobUnsealedBatch {
     /// Which updates are included in this batch.
     pub desc: Description<SeqNo>,
     /// The updates themselves.
-    pub updates: Vec<((Vec<u8>, Vec<u8>), u64, isize)>,
+    pub updates: Vec<ColumnarRecords>,
```

Comment:
weird, looks like I never submitted this comment in the first round. sorry about that, it's my major concern and why I didn't just stamp this PR the first time...

why is this still a Vec? should it not just be a bare ColumnarRecords?

Reply:
Right. That was my expectation when I first started making this change, and I tried to think of ways to make it a bare `ColumnarRecords` but it seemed that, since each write itself produces a `ColumnarRecords`, concatenating together N `ColumnarRecords` into one would introduce a lot more extra allocation / memcpying on the write fast path. I agree that trace batches should just be a bare `ColumnarRecords`

to be entirely honest though I didn't do the performance test of what the impact would be of converting all of the `ColumnarRecords` from various writes into one `ColumnarRecords` per persisted stream and thats probably worth doing before we decide on this approach

Reply:
yeah okay, I see. I'm looking ahead to switching the update blobs to be encoded with arrow where we won't really be able to keep this as a Vec, but I'm fine leaving this as a TODO for then.

Reply:
added a todo here.